# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../netkit-telnet/prototypes.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./telnet/telnet.cc.orig	Fri Sep 24 08:29:33 1999
+++ ./telnet/telnet.cc	Fri Sep 24 08:38:06 1999
@@ -61,6 +61,9 @@
  * Due to lossage in some linux distributions/kernel releases/libc versions
  * this must come *after* termios.h (which is included in externs.h)
  */
+
+extern "C" { extern int tgetent(char *, const char *); }
+
 #include <termcap.h>
 
 #ifdef USE_NCURSES
--- ./telnet/main.cc.orig	Thu May  2 11:37:25 2002
+++ ./telnet/main.cc	Thu May  2 11:39:09 2002
@@ -52,6 +52,8 @@
 #include "defines.h"
 #include "proto.h"
 
+extern "C" { extern void exit(int); }
+
 /*
  * Initialize variables.
  */
--- ./telnet/network.cc.orig	Thu May  2 11:39:50 2002
+++ ./telnet/network.cc	Thu May  2 11:40:00 2002
@@ -49,6 +49,8 @@
 #include "proto.h"
 #include "netlink.h"
 
+extern "C" { extern void exit(int); }
+
 ringbuf netoring;
 ringbuf netiring;
 
--- ./telnet/terminal.cc.orig	Thu May  2 11:40:16 2002
+++ ./telnet/terminal.cc	Thu May  2 11:40:58 2002
@@ -53,6 +53,9 @@
 #include "proto.h"
 #include "terminal.h"
 
+extern "C" { extern void exit(int); }
+extern "C" { extern void *memcpy(void *, const void *, size_t); }
+
 static int TerminalWrite(const char *buf, int n);
 static int TerminalRead(char *buf, int n);
 
--- ./telnet/utilities.cc.orig	Thu May  2 11:41:28 2002
+++ ./telnet/utilities.cc	Thu May  2 11:43:26 2002
@@ -54,6 +54,11 @@
 #include "proto.h"
 #include "terminal.h"
 
+extern "C" { extern void exit(int); }
+extern "C" { extern size_t strlen(const char *); }
+extern "C" { extern char *strcmp(const char *, const char *); }
+extern "C" { extern char *strcpy(char *, const char *); }
+
 FILE *NetTrace = 0;		/* Not in bss, since needs to stay */ /* ? */
 char NetTraceFile[256] = "(standard output)";
 
--- ./telnet/netlink.cc.orig	Thu May  2 11:43:59 2002
+++ ./telnet/netlink.cc	Thu May  2 11:44:04 2002
@@ -12,6 +12,8 @@
 #include "proto.h"
 #include "ring.h"
 
+extern "C" { extern void *memcpy(void *, const void *, size_t); }
+
 /* In Linux, this is an enum */
 #if defined(__linux__) || defined(IPPROTO_IP)
 #define HAS_IPPROTO_IP
